controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```diff
@@ -461,20 +497,19 @@ func (r *Reconciler) getPublishRequestTask(ctx *context.VirtualMachinePublishReq
 // checkPublishRequestStatus checks the publish request task status and mark Uploaded condition.
 // Returns
 // - if a following publish task is needed (true if a publish request should be sent, otherwise return false.),
-// - the uploaded item UUID
 // - error
 // It filters VM Publish task from VC task manager by activation ID and checks its status.
 // - If this task is queued/running, then we should mark Uploaded to false and no need to retry VM publish.
 // - task succeeded, mark Uploaded to true and return the uploaded item ID.
 // - task failed, mark Uploaded to false and retry the operation.
-func (r *Reconciler) checkPublishRequestStatus(ctx *context.VirtualMachinePublishRequestContext) (string, bool, error) {
+func (r *Reconciler) checkPublishRequestStatus(ctx *context.VirtualMachinePublishRequestContext) (bool, error) {
```

Comment:
I would suggest renaming this function to something like `shouldRePublish` to indicate the returned value.

Reply:
If you do, per @dilyar85 's suggestion -- `s/shouldRePublish/shouldRepublish/g`